test/integration/scheduler/preemption_test.go

```diff
@@ -274,7 +274,57 @@ var _ = ginkgo.Describe("Preemption", func() {
 			util.FinishEvictionForWorkloads(ctx, k8sClient, alphaLowWl, betaMidWl)
 
 			util.ExpectWorkloadsToHaveQuotaReservation(ctx, k8sClient, alphaCQ.Name, alphaMidWl)
+			util.ExpectWorkloadsToHaveQuotaReservation(ctx, k8sClient, betaCQ.Name, betaHighWl)
 			util.ExpectWorkloadsToBePending(ctx, k8sClient, alphaLowWl, betaMidWl)
+
+			conditionCmpOpts := cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime")
+			ginkgo.By("Verify the Preempted condition", func() {
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(alphaLowWl), alphaLowWl)).To(gomega.Succeed())
+					g.Expect(apimeta.FindStatusCondition(alphaLowWl.Status.Conditions, kueue.WorkloadPreempted)).To(gomega.BeComparableTo(&metav1.Condition{
+						Type:    kueue.WorkloadPreempted,
+						Status:  metav1.ConditionTrue,
+						Reason:  "InClusterQueue",
+						Message: fmt.Sprintf("Preempted to accommodate a workload (UID: %s) in the ClusterQueue", alphaMidWl.UID),
+					}, conditionCmpOpts))
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(betaMidWl), betaMidWl)).To(gomega.Succeed())
+					g.Expect(apimeta.FindStatusCondition(betaMidWl.Status.Conditions, kueue.WorkloadPreempted)).To(gomega.BeComparableTo(&metav1.Condition{
+						Type:    kueue.WorkloadPreempted,
+						Status:  metav1.ConditionTrue,
+						Reason:  "InCohort",
+						Message: fmt.Sprintf("Preempted to accommodate a workload (UID: %s) in the cohort", alphaMidWl.UID),
+					}, conditionCmpOpts))
+				}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			})
+
+			ginkgo.By("Verify the Preempted condition on re-admission, as the preemptor is finished", func() {
+				util.FinishWorkloads(ctx, k8sClient, alphaMidWl)
+				util.ExpectWorkloadsToHaveQuotaReservation(ctx, k8sClient, alphaCQ.Name, alphaLowWl)
+				util.ExpectWorkloadsToHaveQuotaReservation(ctx, k8sClient, betaCQ.Name, betaMidWl, betaHighWl)
+
+				gomega.Eventually(func(g gomega.Gomega) {
+					g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(alphaLowWl), alphaLowWl)).To(gomega.Succeed())
+					g.Expect(apimeta.FindStatusCondition(alphaLowWl.Status.Conditions, kueue.WorkloadPreempted)).To(gomega.BeComparableTo(&metav1.Condition{
+						Type:    kueue.WorkloadPreempted,
+						Status:  metav1.ConditionFalse,
+						Reason:  "InClusterQueue",
+						Message: fmt.Sprintf("Preempted to accommodate a workload (UID: %s) in the ClusterQueue", alphaMidWl.UID),
```

Comment:
uhm... this is rather confusing, because now the workload isn't preempted.

Should we add a prefix? Like: `Previously: `

Reply:
Ok, so I only flipped status for consistency with how it was done for the "Evicted" condition.

However, I think ideally, the condition should indicate the reason for the "last" transition.
The [API comment](https://github.com/kubernetes/kubernetes/blob/9791f0d1f39f3f1e0796add7833c1059325d5098/pkg/apis/batch/types.go#L641-L643), Also, an example in the Job controller when transitioning from True to False (see [here](https://github.com/kubernetes/kubernetes/blob/9791f0d1f39f3f1e0796add7833c1059325d5098/pkg/controller/job/job_controller.go#L937C128-L937C138).

In this case of transitoning Evicted / Preempted conditions from `True->False`, the reason for this transition is that it has now QuotaReserved. OTOH, it might be handy to still keep the information why the workload was preempted in the first place.

Reply:
Proposed a change along the lines in the last commit. PTAL.

Reply:
> When we see only the message, it may bring any confusion. But doesn't avoid confusion since we have a `Status: False` field?

I have already received tickets from customers that missed the status: false line. So better avoid the confusion.

Reply:
> I have already received tickets from customers that missed the status: false line. So better avoid the confusion.

I see. End-user's feedback is important. Thank you for sharing it with me.
/lgtm